src/harness/definitelyRunner.ts

```diff
@@ -0,0 +1,57 @@
+/// <reference path="harness.ts"/>
+/// <reference path="runnerbase.ts" />
+class DefinitelyTypedRunner extends RunnerBase {
+    private static readonly testDir = "../DefinitelyTyped/types/";
+
+    public workingDirectory = DefinitelyTypedRunner.testDir;
+
+    public enumerateTestFiles() {
+        return Harness.IO.getDirectories(DefinitelyTypedRunner.testDir);
+    }
+
+    public kind(): TestRunnerKind {
+        return "definitely";
+    }
+
+    /** Setup the runner's tests so that they are ready to be executed by the harness
+     *  The first test should be a describe/it block that sets up the harness's compiler instance appropriately
+     */
+    public initializeTests(): void {
+        // Read in and evaluate the test list
+        const testList = this.tests && this.tests.length ? this.tests : this.enumerateTestFiles();
+
+        describe(`${this.kind()} code samples`, () => {
+            for (const test of testList) {
+                this.runTest(test);
+            }
+        });
+    }
+
+    private runTest(directoryName: string) {
+        describe(directoryName, () => {
+            const cp = require("child_process");
+            const path = require("path");
+            const fs = require("fs");
+
+            it("should build successfully", () => {
+                const cwd = path.join(__dirname, "../../", DefinitelyTypedRunner.testDir, directoryName);
+                const timeout = 600000; // 600s = 10 minutes
+                if (fs.existsSync(path.join(cwd, 'package.json'))) {
+                    const stdio = isWorker ? "pipe" : "inherit";
+                    const install = cp.spawnSync(`npm`, ["i"], { cwd, timeout, shell: true, stdio });

```

Comment:
Should we delete a `package-lock.json` before running `npm i`? As is, if a DT package updates a dep, I don't think it'd get tested here.

Reply:
Ugggg. Is there a way not to emit `package-lock.json`? I'd rather do that instead.

Reply:
Nope! Guess I'll have to remove it!